### ISSUE TITLE
Skip line-ending characters conversion if their visualization is enabled

### ISSUE THREAD

**Issue by Darelbi** [notepad-plus-plus/notepad-plus-plus#4220]
### Description of the Issue
By default, Notepad++ when copy pasting code with line endings inside (multi-line code) will automatically convert all line endings to LF. This is ok, but I think it is better not converting line-endings if we are actually showing line ending characters.

### Steps to Reproduce the Issue
* Get on windows a file with CR-LF line endings
* Enable visualization of line endings characters
* Copy paste some code
* Observe the line ending has been converted to LF only

### Expected Behavior
* When visualization of line endings characters is active, no conversion should happens.

### Actual Behavior
* When visualization of line endings characters is active, line endings characters are converted

**Comment by sasumner** [comment]
@Darelbi The way you have stated your scenario is untrue.  At time of paste, the line-ending characters in the to-be-pasted data are all converted to the line-ending format for the paste destination, i.e., the current editor tab buffer.  So, for example, if there are Linux line-endings in the clipboard and the file to receive the paste is a Windows file, the Linux LF will be converted to CRLF by the paste.  That being said, your desire to have no conversion happen when visible line-endings are turned on is OK.

**Comment by Darelbi** [comment]
Then There is a bug in destination detection. My test case Was a unity3D
generatore csharp file where by default all' endings are LF while endings
added by visual studio are CR LF. Editing the file with notepadpp create LF
endings

Il domenica 25 febbraio 2018, sasumner <notifications@github.com> ha
scritto:

**Comment by sasumner** [comment]
@Darelbi So when Notepad++ opens a file, it makes a guess at what the line-endings are for the entire file.  If you start with a file that is mostly LF line endings and then you add some lines with CRLF line endings with Visual Studio....and then finally you open the file in Notepad++, it seems like it is guessing that your file has LF line-endings, so when you paste something in it converts the line endings on what is pasted to LF.  Sounds like correct Notepad++ behavior to me.  Attempting to mix line-ending types in general is asking for trouble, with Notepad++ or with other tools.  Do you have a valid reason for doing what you are trying to do?

**Comment by Darelbi** [comment]
Yes. In using a tool, unity3D, that generate files with Unix line endings,
but code is written with visual studio. Trova get RID of warning i editor
linea endings with notepadpp. Note, if I Press entertainment with notepadpp
it correct add CR LF, but if i copy paste a lineì if code CR LF is
convertendo ti LF, seems inconsistent behaviour too

Il domenica 25 febbraio 2018, sasumner <notifications@github.com> ha
scritto:
